common/containers/Tabs/SendTransaction/components/WalletInfo.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import translate, { translateRaw } from 'translations';
+import { IWallet } from 'libs/wallet';
+import { print } from 'components/PrintableWallet';
+import { Identicon, QRCode } from 'components/ui';
+import GenerateKeystoreModal from 'components/GenerateKeystoreModal';
+import './WalletInfo.scss';
+
+interface Props {
+  wallet: IWallet;
+}
+
+interface State {
+  address: string;
+  privateKey: string;
+  isPrivateKeyVisible: boolean;
+  isKeystoreModalOpen: boolean;
+}
+
+export default class WalletInfo extends React.Component<Props, State> {
+  public state = {
+    address: '',
+    privateKey: '',
+    isPrivateKeyVisible: false,
+    isKeystoreModalOpen: false
+  };
+
+  public componentDidMount() {
+    this.setWalletAsyncState(this.props.wallet);
+  }
+
+  public componentWillReceiveProps(nextProps) {
+    if (this.props.wallet !== nextProps.wallet) {
+      this.setWalletAsyncState(nextProps.wallet);
+    }
+  }
+
+  public render() {
+    const { address, privateKey, isPrivateKeyVisible, isKeystoreModalOpen } = this.state;
+
+    return (
+      <div className="WalletInfo">
+        <div className="Tab-content-pane">
+          <div className="row form-group">
+            <div className="col-xs-11">
+              <label>{translate('x_Address')}</label>
+              <input className="form-control" disabled={true} value={address} />
+            </div>
+            <div className="col-xs-1" style={{ padding: 0 }}>
+              <Identicon address={address} />
+            </div>
+          </div>
+
+          {privateKey && (
+            <div className="row form-group">
+              <div className="col-xs-12">
+                <label>{translate('x_PrivKey')}</label>
+                <div className="input-group">
+                  <input
+                    className="form-control"
+                    disabled={true}
+                    type={isPrivateKeyVisible ? 'text' : 'password'}
+                    value={privateKey}
+                  />
+                  <span
+                    onClick={this.togglePrivateKey}
+                    aria-label={translateRaw('GEN_Aria_2')}
+                    role="button"
+                    className="input-group-addon eye"
+                  />
+                </div>
+              </div>
+            </div>
+          )}
+        </div>
+
+        <div className="Tab-content-pane">
+          <div className="row">
+            <div className="col-xs-6">
+              <label>Public Address</label>
+              <div className="WalletInfo-qr well well-lg">
+                <QRCode data={address} />
+              </div>
+            </div>
+            {privateKey && (
+              <div className="col-xs-6">
+                <label>Private Key</label>
+                <div className="WalletInfo-qr well well-lg" onClick={this.togglePrivateKey}>
+                  <QRCode data={isPrivateKeyVisible ? privateKey : '0'} />
```

Comment:
Would it be better here if instead of checking if privateKey is visible for `QRCode` and not visible for `qr-cover`, you can just render one or the other depending on `isPrivateKeyVisible`? It would involve less boolean logic

Reply:
I need the QR code markup rendered for height sizing unfortunately. I could make some kind of dummy element to fill in the sizing, but if we ever change the QR code component in a way that affected its size or shape, we'd also have to change that dummy element. I figured this was simpler.